modular_citadel/code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,676 @@
+							M.health = M.maxHealth			//Eating fish heals the predator
+				else
+					to_chat(usr, "There are no fish in [src]!")
+		else
+			attack_generic(M, M.harm_intent_damage)
+	else if(istype(M, /mob/living/simple_animal/hostile/bear))
+		if(M.a_intent == INTENT_HELP)							//Bears can try to fish in open tanks on help intent
+			if(lid_switch)									//Can't fish in a closed tank. Fishbowls are ALWAYS open.
+				M.visible_message("[M.name] scrapes it's claws along \the [src]'s lid.", "The lid is closed, so you scrape your claws against \the [src]'s lid.")
+			else
+				if(fish_count)								//Tank must actually have fish to try catching one
+					M.visible_message("[M.name] reaches into  \the[src] and attempts to fish through the opening!", "You reach into \the [src] and begin fishing through the opening!")
+					spawn(5)
+						if(water_level && prob(5))			//Bears are good at catching fish, only a 5% chance to fail
+							M.visible_message("[M.name] swipes at the water!", "You just barely missed that fish!")
+						else								//No water or didn't slip, get that fish!
+							M.visible_message("[M.name] catches and devours a live fish!", "You catch and devour a live fish, yum!")
+							kill_fish()						//Kill a random fish
+							M.health = M.maxHealth			//Eating fish heals the predator
+				else
+					to_chat(usr, "There are no fish in [src]!")
+		else
+			attack_generic(M, M.harm_intent_damage)
+	else
+		if(M.melee_damage_upper > 0)						//If the simple_animal has a melee_damage_upper defined, use that for the damage
+			attack_generic(M, M.melee_damage_upper)
+		else if(M.a_intent == INTENT_HARM)						//Let any simple_animal try to break tanks when on harm intent
+			if(M.harm_intent_damage <= 0) return			//If it doesn't do damage, don't bother with the attack
+			attack_generic(M, M.harm_intent_damage)
+	check_health()
+
+/obj/machinery/fishtank/attack_alien(mob/living/user as mob)
+	if(islarva(user)) return
+	attack_generic(user, 15)
+
+/obj/machinery/fishtank/attack_slime(mob/living/user as mob)
+	attack_generic(user, rand(10, 15))
+
+/obj/machinery/fishtank/attackby(var/obj/item/W, mob/user, params)
+	add_fingerprint(user)
+	if(istype(W,/obj/item/weldingtool) && user.a_intent == INTENT_HELP)
+		var/obj/item/weldingtool/WT = W
+		if(obj_integrity < max_integrity)
+			if(WT.use(5))
+				to_chat(user, "<span class='notice'>You begin repairing [src]...</span>")
+				playsound(src, WT.usesound, 40, 1)
+				if(do_after(user, 40*WT.toolspeed, target = src))
+					obj_integrity = max_integrity
+					playsound(src, 'sound/items/Welder2.ogg', 50, 1)
+					to_chat(user, "<span class='notice'>You repair [src].</span>")
+		else
+			to_chat(user, "<span class='warning'>[src] is already in good condition!</span>")
+		return
+	//Open reagent containers add and remove water
+
+	if(W.is_open_container())
+		if(istype(W, /obj/item/reagent_containers/glass))
+			if(lid_switch)
+				to_chat(usr, "Open the lid on \the [src] first!")
```

Comment:
no usr and when i say no usr it means use user or something instead as usr is not gaurunteed to be set.